test/hotspot/jtreg/compiler/intrinsics/math/TestCopySignIntrinsic.java

```diff
@@ -113,12 +102,7 @@ public void testCopySignF() {
     @Check(test = "testCopySignF")
     public void checkCopySignF() {
         for (int i = 0; i < SIZE; i++) {
-            if (afresult[i] != efresult[i]) {
-                if (Float.isNaN(afresult[i]) ^ Float.isNaN(efresult[i]))  {
-                    throw new RuntimeException("Incorrect result, Math.copySign(" + fmagnitude[i] + " , " + fsign[i] + ") => " +
-                                               "expected : " + efresult[i] + " != " + " actual : " + afresult[i]);
-                }
-            }
+            Verify.checkEQ(afresult[i], efresult[i]);
```

Comment:
You could add a comment that we consider NaN with different encoding as the same value.